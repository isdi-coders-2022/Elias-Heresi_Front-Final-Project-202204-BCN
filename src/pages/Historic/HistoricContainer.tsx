import styled from "styled-components";

export const HistoricContainer = styled.div`
  height: 100vh;
  h2 {
    text-align: center;
    margin-top: 30px;
  }

  h1 {
    padding-top: 25px;
    text-align: center;
  }
  .card {
    margin: 25px 10px 35px 10px;
    margin-left: auto;
    margin-right: auto;
  }
`;
