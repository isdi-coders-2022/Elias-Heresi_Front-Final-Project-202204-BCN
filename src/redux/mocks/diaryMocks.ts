import { DiaryEntry, Diary } from "../interfaces/DiaryInterface";

export const mockFirstEntry = {
  date: new Date("December 17, 2021 03:24:00"),
  vitality: 7,
  positiveEmotions: 5,
  engagement: 9,
  relationships: 2,
  meaning: 4,
  accomplishment: 6,
  wellBeing: 5,
  image: "url",
  commentary: "Today was OK",
} as DiaryEntry;

export const mockSecondEntry = {
  date: new Date("January 17, 2021 03:24:00"),
  vitality: 5,
  positiveEmotions: 7,
  engagement: 2,
  relationships: 7,
  meaning: 9,
  accomplishment: 2,
  wellBeing: 7,
  image: "url",
  commentary: "Today was better",
} as DiaryEntry;

export const mockDiary = {
  username: "champagne_papi",
  diary: [mockFirstEntry, mockSecondEntry],
} as Diary;
