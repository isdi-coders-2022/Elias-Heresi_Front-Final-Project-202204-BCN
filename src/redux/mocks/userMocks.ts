export const usedToken =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiTWFydGEiLCJzdXJuYW1lIjoiQW1pZ2EiLCJ1c2VybmFtZSI6Im1hcnRhIiwiaWF0IjoxNjU0NTAxNTk5fQ.4ygwz7wwvI6b2VVr1Zr58XONUZC9mafuFn6wfBaJ5PA";
