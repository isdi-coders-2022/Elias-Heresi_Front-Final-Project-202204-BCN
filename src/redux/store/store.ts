import { configureStore } from "@reduxjs/toolkit";
import userReducer from "../features/userSlice";
import uiReducer from "../features/uiSlice";
import diaryReducer from "../features/diarySlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    ui: uiReducer,
    diary: diaryReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
