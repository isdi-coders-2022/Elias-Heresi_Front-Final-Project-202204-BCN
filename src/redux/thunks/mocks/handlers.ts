import { rest } from "msw";
export const mockToken: string = "xxx";

export const handlers = [
  rest.post(`${process.env.REACT_APP_API_URL}user/login`, (req, res, ctx) =>
    res(ctx.status(200), ctx.json({ token: mockToken }))
  ),
  rest.post(`${process.env.REACT_APP_API_URL}user/register`, (req, res, ctx) =>
    res(ctx.status(200), ctx.json({ token: mockToken }))
  ),
];
