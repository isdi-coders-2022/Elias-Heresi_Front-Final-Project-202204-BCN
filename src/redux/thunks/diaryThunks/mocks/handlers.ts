import { rest } from "msw";
import { mockApiGetResponse, mockApiId } from "../../../mocks/diaryMocks";

export const handlers = [
  rest.get(`${process.env.REACT_APP_API_URL}diary/all`, (req, res, ctx) =>
    res(ctx.status(201), ctx.json({ entries: mockApiGetResponse }))
  ),
  rest.delete(`${process.env.REACT_APP_API_URL}diary/delete`, (req, res, ctx) =>
    res(ctx.status(201))
  ),
  rest.post(`${process.env.REACT_APP_API_URL}diary`, (req, res, ctx) =>
    res(
      ctx.status(201),
      ctx.json({
        id: mockApiId,
        msg: "The entry was successfully created in marta's diary",
      })
    )
  ),
];
