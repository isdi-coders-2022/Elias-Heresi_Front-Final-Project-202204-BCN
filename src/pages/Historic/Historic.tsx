import { Row, Col } from "react-bootstrap";
import { useSelector } from "react-redux";
import { ToastContainer } from "react-toastify";
import EntrySummary from "../../components/EntrySummary/EntrySummary";
import NavBar from "../../components/NavBar/NavBar";
import { DiaryState } from "../../redux/interfaces/DiaryInterface";
import { RootState } from "../../redux/store/store";
import { notify } from "../../utils/toast";
import { HistoricContainer } from "./HistoricContainer";

const Historic = () => {
  const { collection }: DiaryState = useSelector(
    (state: RootState) => state.diary
  );

  return (
    <>
      <NavBar />
      <ToastContainer />
      <HistoricContainer>
        <Row>
          <h1 onClick={() => notify({ message: "Hola", type: "info" })}>
            Your well-being history
          </h1>
          {collection.length > 0 &&
            collection.map((entry, index) => (
              <Col key={index}>
                <EntrySummary entry={entry} />
              </Col>
            ))}
        </Row>
      </HistoricContainer>
    </>
  );
};

export default Historic;
