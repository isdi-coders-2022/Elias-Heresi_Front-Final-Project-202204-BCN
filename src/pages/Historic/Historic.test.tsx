import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { mockApiGetResponse } from "../../redux/mocks/diaryMocks";
import { usedToken } from "../../redux/mocks/userMocks";
import store from "../../redux/store/store";
import Historic from "./Historic";

let mockCollection = mockApiGetResponse;
let mockPage = 1;
let mockperPage = 1;
let mockTotal = 1;
const mockUsername = "Chocolatero";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => jest.fn(),
  useSelector: () => ({
    collection: mockApiGetResponse,
    loading: false,
    username: "marta",
    name: mockUsername,
    page: 1,
    perPage: 10,
    total: 20,
  }),
}));

const localStorageMock = (() => {
  return {
    getItem() {
      return usedToken;
    },
  };
})();
Object.defineProperty(window, "localStorage", {
  value: localStorageMock,
});

jest.mock("react-chartjs-2", () => ({
  PolarArea: () => <canvas role="img" />,
}));

describe("Given the Historic page component", () => {
  describe("When it's invoked", () => {
    test("Then if the collection contains 1 entries, then 1 card will be rendered", () => {
      const numberOfExpectedEntries = 1;
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Historic />
          </Provider>
        </BrowserRouter>
      );

      const searchedCards = screen.getAllByRole("img");

      expect(searchedCards).toHaveLength(numberOfExpectedEntries);
    });
    test("Then the username will be rendered", () => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Historic />
          </Provider>
        </BrowserRouter>
      );
      const searchedText = screen.getByRole("heading");

      expect(searchedText).toHaveTextContent(mockUsername);
    });
  });
});
