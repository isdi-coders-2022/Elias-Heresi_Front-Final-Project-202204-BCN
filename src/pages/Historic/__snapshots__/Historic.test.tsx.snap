// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Historic page component When it's invoked Then it should always match this snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="#home"
      >
        Bonanza
      </a>
      <button
        aria-controls="/historic"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rr-ui-event-key="/historic"
            disabled={false}
            href="/historic"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/historic"
            disabled={false}
            href="/historic"
            onClick={[Function]}
          >
            Past entries
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="#link"
            disabled={false}
            href="#link"
            onClick={[Function]}
          >
            Know thyself
          </a>
          <div
            className="nav-item dropdown"
          >
            <a
              aria-expanded={false}
              className="dropdown-toggle nav-link"
              data-rr-ui-event-key={null}
              href="#"
              id="basic-nav-dropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span>
                UserÂ 
                <svg
                  color="black"
                  fill="currentColor"
                  height={20}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                  viewBox="0 0 448 512"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="sc-gsnTZi jGMfoV"
  >
    <div
      className="card"
      style={
        Object {
          "width": "30rem",
        }
      }
    >
      <div
        className="no-gutters row"
      >
        <div
          className="col-6"
        >
          <div
            className="sc-bczRLJ icUQoL"
          >
            <canvas
              height={150}
              role="img"
              width={300}
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card-body"
          >
            <div
              className="row"
            >
              <div
                className="col-7"
              >
                <div
                  className="card-title h5"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                >
                  2020-05-18
                </div>
              </div>
              <div
                className="col-2 offset-3"
              >
                <svg
                  color="red"
                  fill="currentColor"
                  height={20}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                  viewBox="0 0 512 512"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                  />
                </svg>
              </div>
            </div>
            <p
              className="card-text"
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                }
              }
            >
              Well being: 
              6
            </p>
            <p
              className="card-text"
            >
              Had a good day at the office. Excited on what is coming...
            </p>
            <div
              className="row"
            >
              <div
                className="offset-2 col"
              >
                <button
                  className="d-flex align-items-center justify-content-center btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  See details
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
